google_bigquery_writer/app.py

```diff
@@ -1,45 +1,46 @@
-# coding=utf-8
 from keboola import docker
 from google_bigquery_writer.exceptions import UserException
 import google_bigquery_writer.writer
 import google.oauth2.credentials
+from google.cloud import bigquery
 import json
 from google_bigquery_writer import schema_mapper
+from google_bigquery_writer.bigquery_client_factory import BigqueryClientFactory
 
 
 class App:
-    def __init__(self, data_dir=None):
+    def __init__(self, data_dir: str = None):
         self.data_dir = data_dir
         self.cfg = docker.Config(self.data_dir)
         self.writer = None
 
     def get_credentials(self):
         oauthapi_data = self.cfg.get_oauthapi_data()
-        credentials = google.oauth2.credentials.Credentials(
+        return google.oauth2.credentials.Credentials(
             oauthapi_data.get('access_token'),
             token_uri='https://accounts.google.com/o/oauth2/token',
             client_id=self.cfg.get_oauthapi_appkey(),
             client_secret=self.cfg.get_oauthapi_appsecret(),
             refresh_token=oauthapi_data.get('refresh_token')
         )
-        return credentials
 
     def get_writer(self):
         """
         Late loading method
         """
         if self.writer:
             return self.writer
-        parameters = self.cfg.get_parameters()
-        my_writer = google_bigquery_writer.writer.Writer(
-            project=parameters.get('project'),
-            credentials=self.get_credentials()
+
+        bigquery_client_factory = BigqueryClientFactory(
+            self.cfg.get_parameters().get('project'),
+            self.get_credentials()
         )
-        self.writer = my_writer
+
+        bigquery_client = bigquery_client_factory.create()
+        self.writer = google_bigquery_writer.writer.Writer(bigquery_client)
         return self.writer
 
     def run(self):
-        parameters = self.cfg.get_parameters()
         action = self.cfg.get_action()
         if action == 'run' or action is None or action == '':
```

Comment:
to None a '' by mela resit metoda get_action() (vyhledove)

Reply:
Souhlasim, ale zatim to musime overovat takto.

Reply:
tak udelej prosim issue :)

Reply:
https://github.com/keboola/python-docker-application/issues/7